src/fmu/dataio/dataio.py

```diff
@@ -710,7 +713,8 @@ def generate_metadata(
             # TODO: This needs refinement: _config_is_valid should be removed
             self.config = global_configuration.roundtrip(self.config)
 
-        obj = self._check_obj_if_file(obj)
+        self._object = self._check_obj_if_file(obj)
```

Comment:
I am a bit unsure of how the object can change class along the way, and the need for this self._object.
Can you eloborate?

Also I think we should change the `self._check_obj_if_file(obj)` method so that it doesn't return anything. Right now it is a bit misleading as it appears that the `obj` is being modified in there, but it is not. We can change it in another PR 🙂

Reply:
In the `generate_export_metadata` function below we take in the `obj` and not the `self._object`, it should probably be changed there as well to be consistent 👍